jetty-server/src/main/java/org/eclipse/jetty/server/session/AbstractSessionDataStore.java

```diff
@@ -159,20 +228,101 @@ public void run()
             throw exception.get();
     }
 
+    @Override
+    public boolean exists(String id) throws Exception
+    {
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        final AtomicReference<Boolean> result = new AtomicReference<>();
```

Comment:
Maybe use a single AtomicReference<Object> here and use instanceof check for the exception?  Still ugly, but in a different way.